setup.ts

```diff
@@ -0,0 +1,395 @@
+        await sleep(DEFAULT_SLEEP_INTERVAL);
+        if (`${publicProps.length}` === '0') {
+            // TODO_CHECK this code isn't being hit, and it's throwing build errors (remove the 2 ts-ignores to highlight the issues)
+            // @ts-ignore
+            await this.democracyNotePreimage(this.alice, this.identityAddRegistrar, [this.alice, registrarAccount]);
+            await sleep(DEFAULT_SLEEP_INTERVAL);
+            // @ts-ignore
+            await this.democracyPropose(this.alice, this.identityAddRegistrar, [this.alice, registrarAccount]);
+            await sleep(DEFAULT_SLEEP_INTERVAL);
+        }
+        let referendumInfo = await this.democracyReferendumInfoOf();
+        // Make sure there is at least one referendum in array
+        while (`${referendumInfo.length}` === '0') {
+            await sleep(DEFAULT_SLEEP_INTERVAL);
+            referendumInfo = await this.democracyReferendumInfoOf();
+        }
+        // Extract latest referendum from given array
+        let lastReferendumInfo = referendumInfo[referendumInfo.length - 1];
+        // Make sure this referendum is `isOngoing` status
+        while (
+            !(lastReferendumInfo.value.toJSON() as {
+                [index: string]: AnyJson;
+            }).isOngoing
+        ) {
+            await sleep(DEFAULT_SLEEP_INTERVAL);
+            referendumInfo = await this.democracyReferendumInfoOf();
+            lastReferendumInfo = referendumInfo[referendumInfo.length - 1];
+        }
+        // Now we can safely vote this proposal
+        await this.democracyVote(this.alice);
+        await sleep(DEFAULT_SLEEP_INTERVAL);
+        /**
+         * query the result of registrar
+         */
+        registrars = await this.identityRegistrars();
+
+        let waiting = true;
+        let regIndex = -1;
+        while (waiting) {
+            await sleep(DEFAULT_SLEEP_INTERVAL);
+            registrars = await this.identityRegistrars();
+            console.log(`Number of existed registrars: ${registrars.length}`);
+            for (let registrar of registrars) {
+                regIndex += 1;
+
+                const account = (registrar.value.toJSON() as {
+                    [index: string]: AnyJson;
+                }).account;
+
+                console.log(`registrar.value: ${account}`, registrar.value);
+                console.log(`registrarAccount: ${registrarAccount.address}`);
+
+                if (`${account}` === `${account2registrar[registrarAccount.address]}`) {
+                    waiting = false;
+                    break;
+                }
+            }
+        }
+
+        /**
+         * set registrar fee and query results
+         */
+        const fee = DEFAULT_REGISTRAR_FEE;
+        await this.identitySetFee(registrarAccount, regIndex, fee);
+        await sleep(DEFAULT_SLEEP_INTERVAL);
+        await this.identityRegistrars();
+        await this.proxyAddProxy(registrarAccount, this.eve.address);
```

Comment:
Consider this line and line 388, it would be duplicated. We can remove this line. 

Let's do `account proxy` in line 387 - line 390